specs/protocol/holocene/exec-engine.md

```diff
@@ -64,19 +63,19 @@ field.
 
 ### Execution
 
-During execution, the EIP-1559 parameters used to calculate the next block base fee should come from the
-parent header's `nonce` field rather than the previous protocol constants, if it is non-zero.
+During execution, the EIP-1559 parameters used to calculate the next block base fee should come from the parent header's
+`extraData` field rather than the previous protocol constants, if it is non-empty.
+
+- If, before Holocene activation, `eip1559Parameters` in the `PayloadAttributesV3` is non-null, the attributes are to be
+  considered invalid by the engine.
 
-- If, before Holocene activation, `eip1559Parameters` in the `PayloadAttributesV3` is non-null, the attributes are to
-  be considered invalid by the engine.
 - At and after Holocene activation:
-  - if `eip1559Params` in the `PayloadAttributesV3` is null, the attributes are to be considered invalid by the
-    engine.
+  - if `eip1559Params` in the `PayloadAttributesV3` is null, the attributes are to be considered invalid by the engine.
   - if `eip1559Params`' `denominator` is `0`, the attributes are to be considered invalid by the engine.
-  - if `parent_header.nonce` is zero, the [canyon base fee parameter constants](../exec-engine.md#1559-parameters) are
-    used for the block's base fee parameters.
-  - if `parent_header.nonce` is non-zero, the EIP-1559 parameters encoded within the parent header's `nonce` field are
-    used for the block's base fee parameters.
+  - if `parent_header.extraData` is empty, the [canyon base fee parameter constants](../exec-engine.md#1559-parameters)
+    are used for the block's base fee parameters.
+  - if `parent_header.extraData` is non-empty with the version byte (first byte) value 0, the remaining 64
+    bytes are used for the block's eip-1559 parameters.
```

Comment:
```suggestion
  - if Holocene is not active in `parent_header.timestamp`, then `parent_header.extraData` may be set to anything valid in pre-Holocene context. The  [canyon base fee parameter constants](../exec-engine.md#1559-parameters), or prior EIP-1559 constants are used for the block's base fee parameters.
    - While not set in regular pre-Holocene blocks, the `extraData` was set in the genesis block of some pre-Holocene chains, and other blocks with pre-Bedrock legacy content. It is thus important to check the hardfork before parsing the `extraData`.
  - if Holocene is active in `parent_header.timestamp`, then `parent_header.extraData` is non-empty, with the following encoding:
    - the first byte is set to `0`, used as version byte for the content.
    - the next 8 bytes are used to encode the 1559 params following the [`eip1559Params`
encoding](#eip1559params-encoding).
    - the `extraData` ends there, at 9 bytes length. The `extraData` has a maximum capacity of 32 bytes (to fit in the L1 beacon-chain `extraData` data-type), and may be modified/extended in future versions.
```

A hardfork should also not retro-actively apply constraints on prior blocks, like the parent-block.
The spec needs to call out that the transition block does not read the 1559 params from the more arbitrary pre-holocene block. 
The previous spec rule would create weird behavior for the block right after genesis, if Holocene wasn't active then yet.

Reply:
thanks, did not realize it could be non-empty in previous cases. updated with some tweaking to your language.